File: 3612-process-string-with-special-operations-i/3612-process-string-with-special-operations-i.java
class Solution {
    public String processStr(String s) {
        StringBuilder ans = new StringBuilder();

        for(char ch : s.toCharArray()){
            if(ch == '*'){
                if(ans.length() > 0)
                    ans.deleteCharAt(ans.length()-1);
                else
                    continue;
            }else if(ch == '#'){
                ans.append(ans);
            }else if(ch == '%'){
                ans.reverse();
            }else{
                ans.append(ch);
            }
        }
        return ans.toString();
    }
}

File: 3612-process-string-with-special-operations-i/3612-process-string-with-special-operations-i.cpp
class Solution {
public:
    string processStr(string s) {
        string ans = "";

        for(char ch : s){
            if(ch == '*'){
                if(ans.length() > 0)
                    ans.pop_back();
            }else if(ch == '#'){
                ans += ans;
            }else if(ch == '%'){
                reverse(ans.begin(), ans.end());
            }else{
                ans += ch;
            }
        }
        return ans;
    }
};